src/util/TchapClientUtils.ts

```diff
@@ -0,0 +1,99 @@
+import BasePlatform from "matrix-react-sdk/src/BasePlatform";
+import { MatrixClientPeg } from "matrix-react-sdk/src/MatrixClientPeg";
+import PlatformPeg from "matrix-react-sdk/src/PlatformPeg";
+
+export default class TchapClientUtils {
+    static SYNC_STORE_NAME = "matrix-js-sdk:riot-web-sync";
+
+    /**
+     * Determine weither the app needs a refresh after loading
+     * @param indexedDB the indexDb interface
+     * @returns Promise(true) if a refresh is needed, Promise(false) in other cases
+     */
+    public static async doesNeedRefresh(indexedDB: IDBFactory): Promise<boolean> {
+        if (!indexedDB) {
+            Promise.reject("indexDb is undefined");
+        }
+        //read version of tchap app in localstorage, if it does start with 4, don't refresh
+        if (TchapClientUtils.getAppVersion() && TchapClientUtils.getAppVersion().startsWith("4")) {
+            return Promise.resolve(false);
+        }
+
+        try {
+            //read version of matrix-js-sdk:riot-web-sync, if it is older than version 4, refresh is needed
```

Comment:
can you document why you do this ? In what cases is this necessary ? why do we save tchap_app_version since there is already storeVersion present ?